packages/material-ui/src/LinearProgress/LinearProgress.js

```diff
@@ -1,183 +1,255 @@
+      (styleProps.variant === 'indeterminate' || styleProps.variant === 'query') &&
+      styles.bar1Indeterminate,
+    [`& .${linearProgressClasses.bar1Determinate}`]:
+      styleProps.variant === 'determinate' && styles.bar1Determinate,
+    [`& .${linearProgressClasses.bar1Buffer}`]:
+      styleProps.variant === 'buffer' && styles.bar1Buffer,
+    [`& .${linearProgressClasses.bar2Indeterminate}`]:
+      (styleProps.variant === 'indeterminate' || styleProps.variant === 'query') &&
+      styles.bar2Indeterminate,
+    [`& .${linearProgressClasses.bar2Buffer}`]:
+      styleProps.variant === 'buffer' && styles.bar2Buffer,
+  });
+};
+
+const useUtilityClasses = (styleProps) => {
+  const { classes, variant, color } = styleProps;
+
+  const slots = {
+    root: ['root', `color${capitalize(color)}`, variant],
+    dashed: ['dashed', `dashedColor${capitalize(color)}`],
+    bar1: [
+      'bar',
+      `barColor${capitalize(color)}`,
+      (variant === 'indeterminate' || variant === 'query') && 'bar1Indeterminate',
+      variant === 'determinate' && 'bar1Determinate',
+      variant === 'buffer' && 'bar1Buffer',
+    ],
+    bar2: [
+      'bar',
+      variant !== 'buffer' && `barColor${capitalize(color)}`,
+      variant === 'buffer' && `color${capitalize(color)}`,
+      (variant === 'indeterminate' || variant === 'query') && 'bar2Indeterminate',
+      variant === 'buffer' && 'bar2Buffer',
+    ],
   };
+
+  return composeClasses(slots, getLinearProgressUtilityClass, classes);
 };
 
+const getColorShade = (theme, color) =>
+  theme.palette.mode === 'light'
+    ? lighten(theme.palette[color].main, 0.62)
+    : darken(theme.palette[color].main, 0.5);
+
+const LinearProgressRoot = experimentalStyled(
+  'span',
+  {},
+  {
+    name: 'MuiLinearProgress',
+    slot: 'Root',
+    overridesResolver,
+  },
+)(({ styleProps, theme }) => ({
+  /* Styles applied to the root element. */
+  position: 'relative',
+  overflow: 'hidden',
+  display: 'block',
+  height: 4,
+  zIndex: 0, // Fix Safari's bug during composition of different paint.
+  '@media print': {
+    colorAdjust: 'exact',
+  },
+  backgroundColor: getColorShade(theme, styleProps.color),
+  /* Styles applied to the root element if `variant="buffer"`. */
+  ...(styleProps.variant === 'buffer' && { backgroundColor: 'transparent' }),
+  /* Styles applied to the root element if `variant="query"`. */
+  ...(styleProps.variant === 'query' && { transform: 'rotate(180deg)' }),
+}));
+
+const LinearProgressDashed = experimentalStyled(
+  'span',
+  {},
+  {
+    name: 'MuiLinearProgress',
+    slot: 'Dashed',
+  },
+)(({ styleProps, theme }) => {
+  const backgroundColor = getColorShade(theme, styleProps.color);
+
+  return {
+    /* Styles applied to the additional bar element if `variant="buffer"`. */
+    position: 'absolute',
+    marginTop: 0,
+    height: '100%',
+    width: '100%',
+    animation: `${bufferKeyframe} 3s infinite linear`,
```

Comment:
See the example with the CircularProgress. It's not flying with styled-components.

Reply:
:+1: thanks, fixed